vm/src/eval.rs

```diff
@@ -277,17 +328,37 @@ pub fn eval_inst(vm: &mut NexusVM<impl Memory>) -> Result<()> {
         ECALL { rd } => {
             RD = rd;
             vm.Z = vm.syscalls.syscall(vm.regs.pc, vm.regs.x, &vm.mem)?;
+            // Profile cycles
+            if vm.Z == SyscallCode::ProfileCycles as u32 {
+                handle_profile_cycles(vm)?;
+            }
         }
         UNIMP => {
             PC = vm.inst.pc;
         }
     }
 
+    // Counts cycles per instruction kind.
+    // In RISC-V:
+    // - Memory instructions are 3 cycles
+    // - Branch instructions are 2 cycles
+    // - System call instruction are 4 cycles
+    // - ALU/ALUI/JAR/LUI/AUIPC instructions are 1 cycle
+    // - Unknown instruction is 0 cycle
+    let cycles = match vm.inst.inst {
+        LOAD { .. } | STORE { .. } => 3,
+        BR { .. } => 2,
+        LUI { .. } | AUIPC { .. } | JAL { .. } | JALR { .. } | ALUI { .. } | ALU { .. } => 1,
+        ECALL { .. } => 4,
```

Comment:
Should also include `EBREAK`, just for future-proofing?

Reply:
updated in https://github.com/nexus-xyz/nexus-zkvm/pull/243/commits/25013349658e9b04816cee6f148b723cd4bae97b 
Added `EBREAK { .. } => 4`